blog-post.md

```diff
@@ -0,0 +1,292 @@
+    for (uint i = 0; i < input.length; i++) {
+        result[i] = transform(input[i]);
+    }
+    return result;
+}
+```
+
+Function `map` receives a memory array formed by elements of type `T` and a function which 
+takes a value of type `T` and returns a `U` value. Notation `(T) -> U` represents the 
+type of functions which has a `T` argument and a `U` result. The function `map` 
+traverses the array by applying its function argument in each element of the input array.
+Such patterns of data structure manipulation can be elegantly coded as 
+polymorphic high-order functions, thus enabling more code reuse and less 
+testing effort.
+
+### Type inference
+
+Core Solidity uses type inference algorithm to reduce syntactic verbosity while 
+maintaining the strong static typing guarantees. In our view, static typing is essential 
+for secure smart contract development. The type inference occurs during compilation and 
+provides complete type safety without explicit annotations.
+
+```
+let constant_value = 42; // Inferred as word
+let heterogeneous_tuple = (address(0), true, 42); // Inferred as (address, bool, word)
+constant_value = heterogeneous_tuple ; // this is a type error!
+```
+
+
+
+### Core Solidity and SAIL
+
+The combination of all these features do form a complete programming language, 
+which we call SAIL (Solidity Abstract Intermediate Language). Core Solidity is 
+built on top of SAIL by providing more specialized syntactic constructs that 
+replicate familiar features from Classic Solidity, alongside a mechanism for 
+"desugaring" them into SAIL equivalent code. But what Core Solidity features 
+will be translated into SAIL?
+
+- Data Locations represented as types: Data locations (e.g., storage, memory) 
+are now a part of a type. This enables the creation of composite types with 
+mixed locations, such as a struct that holds references to storage 
+arrays and memory data, as presented in the next code piece:
+
+```
+struct Broker {
+  total : memory(word) 
+, accounts : storage(array(address, word))
+}
+```
+
+- Static/dynamic array - literals: Classic Solidity limits array literals to static 
+arrays. Core Solidity type system allows literals to adopt the correct type (static 
+or dynamic) from their usage context from the start.
+
+The current Core Solidity prototype support all previous features. In the near 
+future, we plan to develop: 
+
+- Compile time evaluation: Inspired by Zig, we plan to include compile-time code 
+evaluation in Core Solidity. Details on how such feature will work are being 
+discussed by Argot Collective Programming Languages research team.
+
+- Redesigned Module System: Currently, the Core Solidity prototype does not have 
+a proper module system. In our tests, we implemented a import directive which 
+behaves like C's `#include`. The design of a proper module system for Core Solidity 
+are being discussed by Argot Collective Programming Languages research team.
```

Comment:
From our discussions it looks like Core Solidity can live with a module system very similar to what we have in Classic Solidity and there will be no big redesign. The change seems more about solcore prototype specifically, so I'd not mention it here as new a feature to be added. It would fit more in some section about limitations of the prototype (if it has to be mentioned at all).